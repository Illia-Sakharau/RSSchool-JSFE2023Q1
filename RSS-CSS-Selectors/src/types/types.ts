export interface ILevels {
  title: string;
  answer: string;
  htmlMap: string;
  selector: string;
  linkToInfo: string;
}
export interface ICurrentLevel extends ILevels {
  currentLevel: number;
  map: ParsedElementsArray;
}

export interface ICreateElementParam {
  tag: string;
  classes?: string[];
  content?: string;
  id?: string;
}

export interface IParsedElem extends ICreateElementParam {
  tag: Elements;
  target: boolean;
}

export type ParsedElementsArray = (IParsedElem | ParsedElementsArray)[];

export enum Elements {
  rack = 'rack',
  pot = 'pot',
  chamomile = 'chamomile',
  tulip = 'tulip',
}
