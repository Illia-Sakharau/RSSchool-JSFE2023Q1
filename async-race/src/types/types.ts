export interface ICreateElementParam {
  tag: string;
  classes?: string[];
  content?: string;
  id?: string;
}

export interface IRoute {
  path: string;
  view: () => void;
}
