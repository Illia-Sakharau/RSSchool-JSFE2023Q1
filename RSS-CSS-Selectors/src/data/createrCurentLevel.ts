import { parserMapToArray } from '../components/function/parsers';
import { ICurrentLevel } from '../types/types';
import levels from './levels';

export default class {
  private curentLvlInfo: ICurrentLevel = {
    currentLevel: 0,
    map: [],
    description: '',
    answer: '',
    htmlMap: '',
  };
  constructor(private currentLvl: number) {
    this.setCurentLvl(currentLvl);
  }
  public setCurentLvl(currentLvl: number) {
    this.currentLvl = currentLvl;
    this.setCurrentLvlInfo(currentLvl);
  }
  private setCurrentLvlInfo(currentLvl: number) {
    this.curentLvlInfo.currentLevel = currentLvl;
    this.curentLvlInfo.map = parserMapToArray(levels[currentLvl].htmlMap);
    this.curentLvlInfo.description = levels[currentLvl].description;
    this.curentLvlInfo.answer = levels[currentLvl].answer;
    this.curentLvlInfo.htmlMap = levels[currentLvl].htmlMap;
  }
  public getAllInfo() {
    return this.curentLvlInfo;
  }
  public getLevel() {
    return this.curentLvlInfo.currentLevel;
  }
  public getMap() {
    return this.curentLvlInfo.map;
  }
  public getDescription() {
    return this.curentLvlInfo.description;
  }
  public getHtmlMap() {
    return this.curentLvlInfo.htmlMap;
  }
  public getAnswer() {
    return this.curentLvlInfo.answer;
  }
}
