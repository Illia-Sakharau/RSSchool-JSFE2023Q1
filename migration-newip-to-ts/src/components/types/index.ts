export interface IRespSourses {
  status: 'ok' | IError,
  sources: ISourse[],
}

export interface IRespArticles {
  status: 'ok' | IError,
  totalResults: number,
  articles: IArticles,
}

interface ISourse {
  id: string,
  name: string,
  description: string,
  url: string,
  category: string,
  language: string,
  country: string,
}

interface IArticles {
  source: {id: string, name: string},
  author: string,
  title: string,
  description: string,
  url: string,
  urlToImage: string,
  publishedAt: string,
  content: string,
}

interface IError {
  status: string,
  code: string,
  message: string,
}