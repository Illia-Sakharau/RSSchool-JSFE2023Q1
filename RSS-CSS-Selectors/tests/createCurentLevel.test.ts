import createrCurentLevel from '../src/data/createrCurentLevel';
import { ILevels } from '../src/types/types';
import { currentLevel, levels } from './_test-data';

jest.mock('../src/data/levels', () => [
  {
    title: 'Test title',
    answer: 'pot',
    selector: 'test',
    linkToInfo: '#',
    htmlMap: `
        <shelf>
          <rack></rack>
          <pot data-target="true"></pot>
          <chamomile></chamomile>
          <tulip></tulip>
        </shelf>
      `,
  },
]);
const createdLevel = new createrCurentLevel(currentLevel.currentLevel);

describe('Test create current level', () => {
  test('should be defined', () => {
    expect(createdLevel).toBeDefined();
  });
  test('should return current level', () => {
    expect(createdLevel.getLevel()).toEqual(currentLevel.currentLevel);
  });
  test('should return title', () => {
    expect(createdLevel.getTitle()).toEqual(currentLevel.title);
  });
  test('should return map', () => {
    expect(createdLevel.getMap()).toEqual(currentLevel.map);
  });
  test('should return answer', () => {
    expect(createdLevel.getAnswer()).toEqual(currentLevel.answer);
  });
  test('should return selector', () => {
    expect(createdLevel.getSelector()).toEqual(currentLevel.selector);
  });
  test('should return linkToInfo', () => {
    expect(createdLevel.getLinkToInfo()).toEqual(currentLevel.linkToInfo);
  });

  test('should return correct answer', () => {
    expect(createdLevel.getHtmlMap()).toMatchSnapshot();
  });
  test('should return correct answer', () => {
    expect(createdLevel.getAllInfo()).toMatchSnapshot();
  });
});
