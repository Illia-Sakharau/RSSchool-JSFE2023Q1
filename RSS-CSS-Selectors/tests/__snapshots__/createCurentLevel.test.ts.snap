// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test create current level should return correct answer 1`] = `
"
        <shelf>
          <rack></rack>
          <pot data-target="true"></pot>
          <chamomile></chamomile>
          <tulip></tulip>
        </shelf>
      "
`;

exports[`Test create current level should return correct answer 2`] = `
{
  "answer": "pot",
  "currentLevel": 0,
  "htmlMap": "
        <shelf>
          <rack></rack>
          <pot data-target="true"></pot>
          <chamomile></chamomile>
          <tulip></tulip>
        </shelf>
      ",
  "linkToInfo": "#",
  "map": [
    {
      "classes": [],
      "id": "",
      "tag": "rack",
      "target": false,
    },
    {
      "classes": [],
      "id": "",
      "tag": "pot",
      "target": true,
    },
    {
      "classes": [],
      "id": "",
      "tag": "chamomile",
      "target": false,
    },
    {
      "classes": [],
      "id": "",
      "tag": "tulip",
      "target": false,
    },
  ],
  "selector": "test",
  "title": "Test title",
}
`;
